File: entities/qualification.go
package entities

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type DoctorQualification struct {
	QualificationID uuid.UUID      `gorm:"primaryKey" json:"qualification_id"`
	DoctorID        uuid.UUID      `gorm:"not null;index" json:"doctor_id"`
	Degree          string         `gorm:"size:100;not null" json:"degree"`
	Specialization  string         `gorm:"size:255" json:"specialization,omitempty"`
	College         string         `gorm:"size:255;not null" json:"college"`
	University      string         `gorm:"size:255;not null" json:"university"`
	YearCompleted   int            `gorm:"not null" json:"year_completed"`
	CreatedAt       time.Time      `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt       time.Time      `gorm:"autoUpdateTime" json:"updated_at"`
	DeletedAt       gorm.DeletedAt `gorm:"index" json:"-"`
}

func (q *DoctorQualification) BeforeCreate(tx *gorm.DB) (err error) {
	if q.QualificationID == uuid.Nil {
		q.QualificationID = uuid.New()
	}
	return nil
}

File: backend_api/entities/qualification.py
import uuid
from sqlalchemy import Column, String, Integer, DateTime, ForeignKey
from sqlalchemy.sql import func
from database import Base

class DoctorQualification(Base):
    __tablename__ = "doctor_qualifications"

    qualification_id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    doctor_id = Column(String(36), ForeignKey("doctors.id", ondelete="CASCADE"), nullable=False, index=True)
    degree = Column(String(100), nullable=False)
    specialization = Column(String(255), nullable=True)
    college = Column(String(255), nullable=False)
    university = Column(String(255), nullable=False)
    year_completed = Column(Integer, nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())
    deleted_at = Column(DateTime(timezone=True), index=True, nullable=True)
